File: frontend/src/api/alerts.js
import { requestJson } from "./request";

const alertMessages = {
  unavailable: "告警服务暂时不可用，请稍后重试",
  error: "告警数据暂时不可用，请稍后重试",
};

export function fetchAlerts(filters = {}, options = {}) {
  const params = new URLSearchParams();
  for (const key of ["alert_source_id", "state", "severity", "service", "environment", "incident_linked", "observed_from", "observed_to"]) {
    if (filters[key] !== undefined && filters[key] !== null && filters[key] !== "") params.set(key, String(filters[key]));
  }
  if (filters.query?.trim()) params.set("query", filters.query.trim());
  params.set("limit", String(Math.min(Math.max(filters.limit ?? 50, 1), 100)));
  params.set("offset", String(Math.max(filters.offset ?? 0, 0)));
  return requestJson(`/api/v1/alerts?${params.toString()}`, { signal: options.signal }, alertMessages);
}

export function fetchAlertSummary(window = "24h", options = {}) {
  return requestJson(`/api/v1/alerts/summary?window=${encodeURIComponent(window)}`, { signal: options.signal }, alertMessages);
}

export function fetchAlertOverview(alertId, options = {}) {
  return requestJson(`/api/v1/alerts/${encodeURIComponent(alertId)}/overview`, { signal: options.signal }, alertMessages);
}

File: backend/src/incident_intelligence/api/routes/alerts.py
from __future__ import annotations

from datetime import UTC, datetime
from typing import Annotated

from fastapi import APIRouter, Depends, Query
from pydantic import ValidationError

from incident_intelligence.api.dependencies import (
    get_alert_center_service,
    require_manual_actor,
)
from incident_intelligence.api.errors import ApiError
from incident_intelligence.api.schemas.alerts import (
    AlertOverviewResponse,
    AlertPageResponse,
    AlertSummaryResponse,
)
from incident_intelligence.domain.models import Environment, Severity
from incident_intelligence.services.alert_center import (
    AlertCenterService,
    AlertListFilters,
    AlertResourceNotFound,
    AlertStateValue,
    SummaryWindow,
)

router = APIRouter(prefix="/api/v1/alerts", tags=["alerts"])
ManualActor = Annotated[str, Depends(require_manual_actor)]
AlertService = Annotated[AlertCenterService, Depends(get_alert_center_service)]


@router.get("", response_model=AlertPageResponse)
def list_alerts(
    actor: ManualActor,
    service: AlertService,
    alert_source_id: Annotated[str | None, Query()] = None,
    state: Annotated[AlertStateValue | None, Query()] = None,
    severity: Annotated[Severity | None, Query()] = None,
    service_name: Annotated[str | None, Query(alias="service", max_length=128)] = None,
    environment: Annotated[Environment | None, Query()] = None,
    incident_linked: Annotated[bool | None, Query()] = None,
    observed_from: Annotated[datetime | None, Query()] = None,
    observed_to: Annotated[datetime | None, Query()] = None,
    query: Annotated[str | None, Query(max_length=100)] = None,
    limit: Annotated[int, Query(ge=1, le=100)] = 50,
    offset: Annotated[int, Query(ge=0, le=10_000)] = 0,
) -> AlertPageResponse:
    del actor
    try:
        filters = AlertListFilters(
            alert_source_id=alert_source_id,
            state=state,
            severity=severity,
            service=service_name,
            environment=environment,
            incident_linked=incident_linked,
            observed_from=observed_from,
            observed_to=observed_to,
            query=query,
            limit=limit,
            offset=offset,
        )
    except ValidationError as error:
        raise ApiError(422, "validation_error", "告警筛选条件不符合约束") from error
    return AlertPageResponse.model_validate(service.list_alerts(filters).model_dump())


@router.get("/summary", response_model=AlertSummaryResponse)
def summarize_alerts(
    actor: ManualActor,
    service: AlertService,
    window: Annotated[SummaryWindow, Query()] = "24h",
) -> AlertSummaryResponse:
    del actor
    return AlertSummaryResponse.model_validate(
        service.summarize(window, now=datetime.now(UTC)).model_dump()
    )


@router.get("/{alert_id}/overview", response_model=AlertOverviewResponse)
def get_alert_overview(
    alert_id: str,
    actor: ManualActor,
    service: AlertService,
) -> AlertOverviewResponse:
    del actor
    try:
        result = service.get_overview(alert_id)
    except AlertResourceNotFound as error:
        raise ApiError(404, "resource_not_found", "未找到指定告警") from error
    return AlertOverviewResponse.model_validate(result.model_dump())
